### PR TITLE
Added pop method on collection, collection-proxy, and test

### LINKED ISSUE #37: Add "pop" method
The `collection.pop()` method removes and returns the last item from the collection:

**Example**

```js
const collection = Collect([4, 1, 37, 2, 12])

await collection.pop()

// 12

await collection.all()

// [4, 1, 37, 2]
```

### PR DESCRIPTION
**Pull request by sanskagarwal**
Closes #37 
Added Tests

### CODE REVIEW

File: test/collection.js [hunk 1]
```diff
@@ -628,6 +628,18 @@ describe('Chained Collection', () => {
...
+    expect(
+      await Collect([1, 2, 3])
+        .pop()
+    ).to.equal(3)
+
+    expect(
+      await Collect([])
+        .pop()
+    ).to.equal(undefined)
+  })
```
Comment (marcuspoehls) [inline test/collection.js:646]: Can you please add a test that checks if the original array is changed (the last item removed)?

**Review by marcuspoehls — CHANGES_REQUESTED** [review]
@skbro Hey Sanskar, thank you for the PR! Looks good. I’m requesting an additional test. Can you please have a look? Thank you!

File: test/collection.js (hunk 1 above)
Comment (sanskagarwal) [reply test/collection.js:646]: Sure

**Review by marcuspoehls — APPROVED** [review]

**Comment by marcuspoehls** [comment]
@skbro Thank you! Merged!

**Comment by sanskagarwal** [comment]
Thank you for your time :)

**Pull request merged**